tasks/insights-client.yml

```diff
@@ -10,6 +10,25 @@
   when:
     - rhc_state | d("present") != "reconnect"
 
+- name: Add insights tags
+  copy:
+    dest: /etc/insights-client/tags.yml
+    content: "{{ rhc_insights.tags | to_nice_yaml(indent=2) }}"
+    mode: '0644'
+  register: __insights_tags_added
+  when:
+    - rhc_insights.tags != None
+    - rhc_insights.tags | d({}) | length > 0
+
+- name: Delete insights tags
+  file:
+    state: absent
+    path: /etc/insights-client/tags.yml
+  register: __insights_tags_removed
+  when:
+    - rhc_insights.tags != None
+    - rhc_insights.tags | d({}) | length == 0
```

Comment:
This is a little bit tricky - you want to remove the file if, and only if, `rhc_insights.tags == {}` - so perhaps
`when: rhc_insights.tags | d(None) == {}`
in this case `rhc_insights.tags` must be explicitly defined to the value `{}` - otherwise, the file is not removed - otherwise, cases like `rhc_insights.tags: []` or `rhc_insights.tags: ""` will cause the file to be removed since the length is 0.

In general, it is better to be explicit than implicit.